packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -387,64 +415,123 @@ export const splitFormatIntoSections = <TDate>(
   };
 
   for (let i = 0; i < expandedFormat.length; i += 1) {
+    const escapedPartOfCurrentChar = escapedParts.find(
+      (escapeIndex) => escapeIndex.start <= i && escapeIndex.end >= i,
+    );
+
     const char = expandedFormat[i];
 
-    if (char.match(/([A-zÀ-ú]+)/g)) {
+    if (escapedPartOfCurrentChar == null && char.match(/([A-Za-z]+)/g)) {
       currentTokenValue += char;
-    } else {
+    }
+    // If we are on the starting of the ending character of an escaped part of the format,
+    // Then we ignore this character.
+    else if (
+      escapedPartOfCurrentChar == null ||
+      (escapedPartOfCurrentChar.start !== i && escapedPartOfCurrentChar.end !== i)
```

Comment:
It's a bit hard to get this logic expression. Maybe would be easier with dedicated varaibles such as 

```js
const isEscapedChar = escapedPartOfCurrentChar != null
// We ignore open/close token of escaped sections
const isIgnoredChar = isEscapedChar && (escapedPartOfCurrentChar.start === i || escapedPartOfCurrentChar.end === i)

// ...

else if (!isIgnoredChar){
  // ..
}
```